### DISCUSSION TITLE
How to opt out persist on some values

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2762] [General]
So I switched from having many store for each feature like userStore, cartStore ... to the recommended  flux way so into having slices and single store. Now previously I could persist which store i wanted independently. So maybe one feature needed to be persisted and other not so it made all sense. But now as per the docs its recommended to only have the middleware on the main store. 

Now the problem with me is that since when persisting data it will also seralize it. It works for the most data types but something like the Date object will be converted to a string and not be the same anymore. So my question if there is a way to not maybe persist some value is possible. Or maybe if provided by the lib use a different storing mechanism so that serialization problems not exsit.

**Comment by dai-shi** [comment]
You can use `partialize` option or a custom `storage`.

**Reply by ImamJanjua** [reply]
I I have already read some discussions about having a custom storage but I am looking for an easier solution. Could u maybe give refrence to docs or other to tell what u mean my partialize.

**Reply by dbritto-dev** [reply]
@ImamJanjua `persist` middleware supports `partialize` where you can define which fields you want to persist instead of whole store
